From a1d38d78163d84d3506c188e195cd5fa363f4be6 Mon Sep 17 00:00:00 2001
From: Adam Duskett <adam.duskett@amarulasolutions.com>
Date: Thu, 17 Aug 2023 13:00:07 -0600
Subject: [PATCH] remove GetStorage

Commit 287b20a8bfc71196cd733625e622b98b2f84bef1 introduced the get_storage
plugin which breaks with the following error when the application is ran:

Unhandled Exception: MissingPluginException(No implementation found for method
getApplicataionDocumentsDirectory on channel plugins.flutter.io/path_provider)

Revert the change.

Upstream: https://github.com/flutter/gallery/issues/994
Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
---
 lib/feature_discovery/feature_discovery.dart | 10 ----------
 lib/main.dart                                |  2 --
 pubspec.yaml                                 |  1 -
 3 files changed, 13 deletions(-)

diff --git a/lib/feature_discovery/feature_discovery.dart b/lib/feature_discovery/feature_discovery.dart
index 288c78b..965d83c 100644
--- a/lib/feature_discovery/feature_discovery.dart
+++ b/lib/feature_discovery/feature_discovery.dart
@@ -6,7 +6,6 @@ import 'package:flutter/material.dart';
 import 'package:flutter/scheduler.dart';
 import 'package:gallery/feature_discovery/animation.dart';
 import 'package:gallery/feature_discovery/overlay.dart';
-import 'package:get_storage/get_storage.dart';
 
 const _featureHighlightShownKey = 'feature_highlight_shown';
 
@@ -271,15 +270,6 @@ class _FeatureDiscoveryState extends State<FeatureDiscovery>
 
     initAnimationControllers();
     initAnimations();
-
-    final localStorage = GetStorage();
-    final featureHiglightShown =
-        localStorage.read<bool>(_featureHighlightShownKey) ?? false;
-    localStorage.write(_featureHighlightShownKey, true);
-    showOverlay = widget.showOverlay && !featureHiglightShown;
-    if (showOverlay) {
-      localStorage.write(_featureHighlightShownKey, true);
-    }
   }
 
   void initAnimationControllers() {
diff --git a/lib/main.dart b/lib/main.dart
index e9f4ff9..8c7a4e3 100644
--- a/lib/main.dart
+++ b/lib/main.dart
@@ -16,7 +16,6 @@ import 'package:gallery/pages/backdrop.dart';
 import 'package:gallery/pages/splash.dart';
 import 'package:gallery/routes.dart';
 import 'package:gallery/themes/gallery_theme_data.dart';
-import 'package:get_storage/get_storage.dart';
 import 'package:google_fonts/google_fonts.dart';
 
 import 'firebase_options.dart';
@@ -26,7 +25,6 @@ export 'package:gallery/data/demos.dart' show pumpDeferredLibraries;
 
 void main() async {
   GoogleFonts.config.allowRuntimeFetching = false;
-  await GetStorage.init();
 
   if (defaultTargetPlatform != TargetPlatform.linux &&
       defaultTargetPlatform != TargetPlatform.windows &&
diff --git a/pubspec.yaml b/pubspec.yaml
index 964edad..4b00e40 100644
--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -24,7 +24,6 @@ dependencies:
   firebase_core: ^2.7.0
   firebase_crashlytics: ^3.1.1
   firebase_performance: ^0.9.0+14
-  get_storage: ^2.1.1
   google_fonts: ^5.0.0
   intl: any # An exact version pin will be provided by the Flutter SDK
   meta: ^1.7.0
-- 
2.41.0

